lib/core/src/rcMisc.cpp

```diff
@@ -4594,12 +4607,42 @@ int get_canonical_name(const char *_hostname, char* _buf, size_t _len) {
     if (ret_getaddrinfo_with_retry ) {
         return ret_getaddrinfo_with_retry;
     }
+
     snprintf(_buf, _len, "%s", p_addrinfo->ai_canonname);
+
+    if (CLIENT_PT != ::ProcessType) {
+        rodsLog(LOG_DEBUG, "%s :: Caching addrinfo for [hostname=%s].", __func__, _hostname);
+
+        //try {
+            const auto age = irods::get_server_property<std::chrono::seconds>(irods::CFG_EVICTION_AGE_IN_SECONDS_KW);
+            net::dnsc_insert_or_assign(target, *p_addrinfo, age);
+        //}
+        //catch (...) {
+            // TODO Should the default be specified in the catch statement.
+            //net::dnsc_insert_or_assign(target, *p_addrinfo, std::chrono::seconds{30});
+        //}
```

Comment:
the outcome of these lines depend on how we want to set the eviction property. see rodsServer.cpp.